pkg/ebpf/bytecode/runtime/asset.go

```diff
@@ -78,22 +90,61 @@ func (a *asset) Compile(config *ebpf.Config, additionalFlags []string, client st
 		return nil, fmt.Errorf("unable to create compiler output directory %s: %w", outputDir, err)
 	}
 
-	protectedFile, err := createProtectedFile(fmt.Sprintf("%s-%s", a.filename, a.hash), outputDir, f)
+	diskProtectedFile, err := createProtectedFile(fmt.Sprintf("%s-%s", a.filename, a.hash), outputDir, f)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create ram backed file from %s: %w", f.Name(), err)
 	}
 	defer func() {
-		if err := protectedFile.Close(); err != nil {
-			log.Debugf("error closing protected file %s: %s", protectedFile.Name(), err)
+		if err := diskProtectedFile.Close(); err != nil {
+			log.Debugf("error closing protected file %s: %s", diskProtectedFile.Name(), err)
 		}
 	}()
+	protectedFile := diskProtectedFile
+	hash := a.hash
 
-	if err = a.verify(protectedFile); err != nil {
-		a.tm.compilationResult = verificationError
+	if err = a.verify(diskProtectedFile); err != nil {
 		return nil, fmt.Errorf("error reading input file: %s", err)
 	}
 
-	out, result, err := compileToObjectFile(protectedFile.Name(), outputDir, a.filename, a.hash, additionalFlags, kernelHeaders)
+	a.tm.compilationResult = compilationErr
+	if modifyCB != nil {
```

Comment:
There's this verify check before checking if `modifyCB` is nil. I'm not sure why exactly but the verify is failing.